src/components/WalletSelector/AddWalletPending.tsx

```diff
@@ -0,0 +1,152 @@
+import Spacer from 'components/core/Spacer/Spacer';
+import { ADDRESS_ALREADY_CONNECTED, INITIAL, CONFIRM_ADDRESS, PROMPT_SIGNATURE } from 'types/Wallet';
+import { convertWalletName } from 'utils/wallet';
+import { useModal } from 'contexts/modal/ModalContext';
+import ManageWalletsModal from 'scenes/Modals/ManageWalletsModal';
+import { initializeAddWalletPipeline } from './authRequestUtils';
+
+type Props = {
+  pendingWalletName: string;
+  pendingWallet: AbstractConnector;
+  setDetectedError: (error: Web3Error) => void;
+};
+
+type PendingState = typeof INITIAL | typeof ADDRESS_ALREADY_CONNECTED | typeof CONFIRM_ADDRESS | typeof PROMPT_SIGNATURE;
+
+// This Pending screen is dislayed after the connector has been activated, while we wait for a signature
+function AddWalletPending({ pendingWallet, pendingWalletName, setDetectedError }: Props) {
+  const {
+    library,
+    account,
+  } = useWeb3React<Web3Provider>();
+  const signer = useMemo(() => library && account ? library.getSigner(account) : undefined, [library, account]);
+
+  const [pendingState, setPendingState] = useState<PendingState>(INITIAL);
+
+  const fetcher = useFetcher();
+  const authenticatedUserAddresses = useAuthenticatedUserAddresses();
+
+  const { showModal } = useModal();
+
+  const openManageWalletsModal = useCallback((address: string) => {
+    showModal(<ManageWalletsModal newAddress={address}/>);
+  }, [showModal]);
+
+  const attemptAddWallet = useCallback(async (address: string, signer: JsonRpcSigner) => {
+    try {
+      const { signatureValid } = await initializeAddWalletPipeline({
+        address,
+        signer,
+        fetcher,
+        connector: pendingWallet,
+      });
+      openManageWalletsModal(address);
+
+      return signatureValid;
+    } catch (error: unknown) {
+      if (isWeb3Error(error)) {
+        setDetectedError(error);
+      }
+
+      // Fall back to generic error message
+      if (error instanceof Error) {
+        const web3Error: Web3Error = { code: 'AUTHENTICATION_ERROR', ...error };
+        setDetectedError(web3Error);
+      }
+    }
+  }, [fetcher, openManageWalletsModal, pendingWallet, setDetectedError]);
+
+  const isMetamask = useMemo(() => pendingWalletName.toLowerCase() === 'metamask', [pendingWalletName]);
+
+  const userFriendlyWalletName = useMemo(() => convertWalletName(pendingWalletName), [pendingWalletName]);
+
+  useEffect(() => {
+    async function authenticate() {
+      if (account && signer) {
+        if (authenticatedUserAddresses.includes(account.toLowerCase())) {
+          setPendingState(ADDRESS_ALREADY_CONNECTED);
+          return;
+        }
+
+        if (isMetamask) {
+          setPendingState(CONFIRM_ADDRESS);
+        } else {
+          await attemptAddWallet(account.toLowerCase(), signer);
+        }
+      }
+    }
+
+    void authenticate();
+  }, [account, signer, authenticatedUserAddresses, attemptAddWallet, isMetamask]);
+
+  if (pendingState === ADDRESS_ALREADY_CONNECTED) {
```

Comment:
add `&& account` so we don't have to optional chain later

Reply:
done ✅